### PR TITLE
ADDITION: Gotify

### PR DESCRIPTION
**Pull request by bcurran3**
**What this PR does / why we need it**:

Add Gotify app.

**Which issue (if any) this PR fixes**:

Fixes # https://github.com/davestephens/ansible-nas/issues/65

**Any other useful info**:

Nice for server specific messages to SysOp.

### CODE REVIEW

**Comment by bcurran3** [comment]
I don't think this is a "me thing."

> ERROR! Unable to retrieve file contents
> Could not find or access '/etc/ansible/playbooks/playbook_under_test/tasks/gotify.yml' on the Ansible Controller.

**Comment by bcurran3** [comment]
Found the problem. Duh!

The rage shall continue.

File: docs/applications/gotify.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Gotify is a simple server for sending and receiving messages.
+
+## Usage
+
+Set `gotify_enabled: true` in your `inventories/<your_inventory>/nas.yml` file.
+
+If you want to change the default port, edit gotify_port_http:.
+
+The Gotify web interface can be found at http://ansible_nas_host_or_ip:8077 and http://ansible_nas_host_or_ip:8078 (internally).
```
Comment (davestephens) [inline docs/applications/gotify.md:15]: What does "8078 internally" mean, isn't that the https port?

File: group_vars/all.yml [hunk 2]
```diff
@@ -823,4 +826,14 @@ vd_users:
...
\ No newline at end of file
+vd_rdp_port: 3389
+
+###
+### Gotify
+###
+gotify_available_externally: "false"
+gotify_data_directory: "{{ docker_home }}/gotify/config"
+gotify_port_http: "8077"
+gotify_port_https: "8078"
```
Comment (davestephens) [inline group_vars/all.yml:837]: This port doesn't appear to be mapped to the container?

File: tasks/gotify.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+- name: Gotify Directory
+  file:
+    path: "{{ item }}"
+    state: directory
+  with_items:
+    - "{{ gotify_data_directory }}/app/data"
```
Comment (davestephens) [inline tasks/gotify.yml:7]: It feels a little clunky adding `/app/data` to the end of the path, as data is then stored in `{{ docker_home }}/gotify/config/app/data`

File: tasks/gotify.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+- name: Gotify Docker Container
+  docker_container:
+    name: gotify
+    image: gotify/server:latest
+    pull: true
+    volumes:
+      - "{{ gotify_data_directory }}/data:/app/data:rw"
```
Comment (davestephens) [inline tasks/gotify.yml:15]: Should drop `/data` from the path per my comment above

**Review by davestephens — CHANGES_REQUESTED** [review]
Looking forward to getting this one merged :+1: Needs a little bit of tweaking then we're good to go.

File: tasks/gotify.yml (hunk 3 above)
Comment (bcurran3) [reply tasks/gotify.yml:7]: Intentional. (Like the other reference.) I like to make the dirs match in the config directory to the container. Personal preference only and I'm not highly committed to it.

File: group_vars/all.yml (hunk 2 above)
Comment (bcurran3) [reply group_vars/all.yml:837]: Fixed - yeah this was pushed when I didn't have Traefik setup yet. :)

**Comment by bcurran3** [comment]
Good now IMHO thought you might still want a simple change.
This one's for you - me giving back.

File: tasks/gotify.yml (hunk 3 above)
Comment (bcurran3) [reply tasks/gotify.yml:7]: BTW: NO config dir.
I'm going to start getting away from the template docker_data/app/config default and go with docker_data/app as the "config"/"data" dir. I really like matching the container's absolute path to a path under docker_data/app - visually much easier to see what's going on and debug when needed.

**Comment by davestephens** [comment]
Closing in favour of #429 as discussed

**Pull request closed without merging**